### PR TITLE
Install user private ssh keys if they are provided in secrets

### PR DESCRIPTION
**Pull request by gandelman-a**
Allows optionally storing user private ssh keys in the secrets
file. This is specifically useful for getting the bastion cideploy
user setup to access other nodes for its ansible deployment.

Resolves: #33

### CODE REVIEW

**Comment by gandelman-a** [comment]
hangon, need to fix this

File: roles/bastion/tasks/main.yml [hunk 1]
```diff
@@ -1,29 +1,63 @@
...
-  file: state=directory path=~{{ item }}/.ssh owner={{ item }} group={{ item }} mode=0700
+  file:
+    state: directory
+    path: "~{{ item }}/.ssh"
+    owner: "{{ item }}"
+    group: "{{ item }}"
+    mode: 0700
+  with_items: "{{ bastion_users }}"
+
+- name: Add SSH public key
```
Comment (nibalizer) [inline roles/bastion/tasks/main.yml]: This is the authorized keys file

File: roles/bastion/tasks/main.yml [hunk 2]
```diff
@@ -1,29 +1,63 @@
...
+    src: "{{ item }}.authorized_keys"
+    dest: "~{{ item }}/.ssh/authorized_keys"
+    owner: "{{ item }}"
+    group: "{{ item }}"
+    mode: 0664
   with_items: "{{ bastion_users }}"
 
-- name: Add SSH key
-  copy: src={{ item }}.authorized_keys dest=~{{ item }}/.ssh/authorized_keys owner={{ item }} group={{ item }} mode=0664
+- name: Add SSH private key
```
Comment (nibalizer) [inline roles/bastion/tasks/main.yml]: We need a public key file stanza that looks like this for the id_rsa.pub

File: roles/bastion/tasks/main.yml (hunk 1 above)
Comment (gandelman-a) [inline roles/bastion/tasks/main.yml]: updated the name here to be clear

File: roles/bastion/tasks/main.yml (hunk 2 above)
Comment (gandelman-a) [inline roles/bastion/tasks/main.yml]: it shouldn't be required that the public key is installed to use the private?

Comment (nibalizer) [inline roles/bastion/tasks/main.yml]: It's not required, but it makes it easier for someone debugging to figure out which key is in use. I can never remember the command to get a public key from a private key, and the comment is lost as well.

**Pull request closed without merging**